src/definitions.jl

```diff
@@ -683,10 +683,11 @@ function adjoint_mul(p::Plan{T}, x::AbstractArray, ::FFTAdjointStyle) where {T}
     pinv = inv(p)
     # Optimization: when pinv is a ScaledPlan, check if we can avoid a loop over x.
     # Even if not, ensure that we do only one pass by combining the normalization with the plan.
+    RT = AbstractArray{<:T} # canonicalize eltype of returned array to be extra safe about type stability  
     if pinv isa ScaledPlan && pinv.scale == N
-        return pinv.p * x
+        return convert(RT, pinv.p * x)
     else
-        return (1/N * pinv) * x
+        return convert(RT, (inv(N) * pinv) * x)
```

Comment:
See my comment here for why I think my initial concern was unjustified: https://github.com/JuliaMath/AbstractFFTs.jl/pull/113/files#r1367787821

However, there could be actual type stability issues in cases where `pinv` is a `ScaledPlan` where `pinv.scale` is not of type `eltype(pinv.p)`. This case is actually intended in the design of `ScaledPlan` and documented in its constructor. It does not matter as long as `typeof(pinv.scale)` does not cause promotions of `eltype(pinv.p)` (such as in the intended case of `Float64` and `Complex{Float64}`), but such a non-promoting behaviour is not guaranteed anywhere in the definition of `ScaledPlan`. Hence I wonder if it's worth it at all to introduce this branch here. Did you perform any benchmarks and see significant speed-ups in practical scenarios?

So maybe we could just simplify this to
```suggestion
  return (inv(N) * pinv) * x
```

Reply:
Agreed: we can keep the one pass optimization but avoid the zero-pass optimization that involves this tricky runtime branching.